api/src/main/java/jakarta/persistence/QueryReference.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Gavin King      - 4.0
+
+package jakarta.persistence;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A reference to a named query declared via the {@link NamedQuery}
+ * or {@link NamedNativeQuery} annotations.
+ *
+ * @see EntityHandler#createQuery(QueryReference)
+ *
+ * @since 4.0
+ */
+public interface QueryReference {
+    /**
+     * The name of the query.
+     */
+    String getName();
+
+    /**
+     * A map keyed by hint name of all hints specified via
+     * {@link NamedQuery#hints} or {@link NamedNativeQuery#hints}.
+     */
+    Map<String,Object> getHints();
+
+    /**
+     * The types of the supplied
+     * {@linkplain #getArguments arguments} to query
+     * parameters, or {@code null} if no arguments were
+     * supplied. Arguments are present when this is a
+     * reference to a query declared using an annotation
+     * of a method.
+     * <p>
+     * Any mutation of the returned list results in an
+     * {@link UnsupportedOperationException}.
+     *
+     * @since 4.0
+     */
+    List<Class<?>> getParameterTypes();
+
+    /**
+     * The names assigned to the supplied
+     * {@linkplain #getArguments arguments} to query
+     * parameters, or {@code null} if no arguments were
+     * supplied. Arguments are present when this is a
+     * reference to a query declared using an annotation
+     * of a method. If the query has named parameters,
+     * these are interpreted as the parameter names.
+     * Otherwise, if the query has positional parameters,
+     * they are ignored.
+     * <p>
+     * Any mutation of the returned list results in an
+     * {@link UnsupportedOperationException}.
+     *
+     * @since 4.0
+     */
+    List<String> getParameterNames();
+
+    /**
+     * The arguments supplied to the query parameters,
+     * or {@code null} if no arguments were supplied.
```

Comment:
Would it be better to use the empty list for no arguments rather than a `null` list?
```suggestion
     * or the {@linkplain java.util.Collections#emptyList empty list}
     * if no arguments were supplied.
```